helix-core/src/main/java/org/apache/helix/manager/zk/CallbackHandler.java

```diff
@@ -319,13 +320,12 @@ public void enqueueTask(NotificationContext changeContext) throws Exception {
         logger.info("CallbackHandler {} is not ready, ignore change callback from path: {}, for "
             + "listener: {}", _uid, _path, _listener);
       } else {
-        synchronized (this) {
-          if (_batchCallbackProcessor != null) {
-            _batchCallbackProcessor.queueEvent(changeContext.getType(), changeContext);
-          } else {
-            throw new HelixException(
-                "Failed to process callback in batch mode. Batch Callback Processor does not exist.");
-          }
+        CallbackProcessor callbackProcessor = _batchCallbackProcessorRef.get();
```

Comment:
Not necessarily wrong behavior. But this may change the behavior.
The older logic will throw an exception if the event won't be processed. But with the new code, it is possible that we add the event to a closed processor without throwing exception?

Reply:
TFTR. Yes, it is a behavior change. However, I think in the old code when er get exception notice like this, we would treat it as a no-harm exception... So I think this change is ok.